src/useCookieConsent.ts

```diff
@@ -31,7 +31,7 @@ export const useCookieConsent = (
   }, [consent]);
 
   const acceptCookies = (newConsent: CookieConsent) => {
-    setConsent(newConsent);
+    setConsent((currentConsent) => ({...currentConsent, ...newConsent}));
```

Comment:
@bring-shrubbery Btw, I was just thinking and this proposal will persist always the current state, maybe another way of only persisting the `necessary` could be something like:
```js
setConsent(({ necessary }) => ({...necessary, ...newConsent}));
```

But let me know your thoughts!

Reply:
Yeah, I think we want for the default behavior to set necessary to `true`, but also have an option for users to override that. Maybe something like this? (options type would have to be placed in `types.ts` file I think)

```ts
const acceptCookies = (newConsent: CookieConsent, options?: { allowChangingNecessary: boolean }) => {
  if (options?.allowChangingNecessary) {
    setConsent(newConsent);
  } else {
    setConsent({...newConsent, necessary: true});
  }
};
```

Reply:
What do you think @edgarlr ?

Reply:
@bring-shrubbery Makes total sense! I'll update the PR with these changes!

Reply:
@bring-shrubbery Done! I just updated the PR, let me know what do you think!

Reply:
Looks good! I think we can merge, just need to figure out what's up with the codecov token